logs_sdk/lib/opentelemetry/sdk/logs/log_record.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # Implementation of OpenTelemetry::Logs::LogRecord that records log events.
+      class LogRecord < OpenTelemetry::Logs::LogRecord
+        attr_accessor :timestamp,
+                      :observed_timestamp,
+                      :severity_text,
+                      :severity_number,
+                      :body,
+                      :attributes,
+                      :trace_id,
+                      :span_id,
+                      :trace_flags,
+                      :resource,
+                      :instrumentation_scope
+
+        # Creates a new {LogRecord}.
+        #
+        # @param [optional Float, Time] timestamp Time when the event occurred.
+        # @param [optional Float, Time] observed_timestamp Time when the event
+        #   was observed by the collection system. If nil, will first attempt
+        #   to set to `timestamp`. If `timestamp` is nil, will set to
+        #   `Process.clock_gettime(Process::CLOCK_REALTIME, :nanosecond)`.
```

Comment:
Why do these accept `Float`? They should only accept `Time`, like corresponding methods in the tracing API. This makes it easier to ensure the correctness of conversion to "Integer nanoseconds since Epoch" internally, and reduces confusion about the units.

Reply:
I got a little too overzealous when trying to adhere to the spec. The datatype in the Logs Data Model was listed as "Type: Timestamp, uint64 nanoseconds since Unix epoch." Since the timestamp could be efficiently accessed using `Process.clock_gettime(Process::CLOCK_REALTIME, :nanosecond)`, I decided to add it and make it the default.

Based on these questions, I decided to remove the `Float` option, go all in on the `Time` object and transform it into a nanosecond `Integer` in `LogRecord#to_log_record_data`. 

Fixes across two commits: 
* d3607592e8127ec0e6f0ca719a7d52649e7b5a5e 
* 0fb4606e20c8d71d4f2c47a71ac59898f4deb2e1